### PR TITLE
Adding rosros to noetic and foxy, and to rosdep/python

### PR DESCRIPTION
**Pull request by suurjaak**
Please add the following dependency to the rosdep database.

## Package name:

`rosros`

## Package Upstream Source:

https://github.com/suurjaak/rosros

## Purpose of using this:

rosros is a wrapper library for providing a convenient unified interface to ROS1 / ROS2 Python API.

Distro packaging links:

## Links to Distribution Packages

- Pip:
  - https://pypi.org/project/rosros

# Please Add This Package to be indexed in the rosdistro.

noetic, foxy

# The source is here:

https://github.com/suurjaak/rosros

# Checks
 - [X] All packages have a declared license in the package.xml
 - [X] This repository has a LICENSE file
 - [X] This package is expected to build on the submitted rosdistro

### CODE REVIEW

File: foxy/distribution.yaml [hunk 1]
```diff
@@ -5710,6 +5710,21 @@ repositories:
       url: https://github.com/uos/rospy_message_converter.git
       version: foxy
     status: maintained
+  rosros:
```
Comment (clalancette) [inline foxy/distribution.yaml:5713]: I really don't think we should have a project called `rosros` in the official distribution.  It is far too generic, and unclear on what it does.  Please consider renaming this to be more descriptive of what it does.  See https://www.ros.org/reps/rep-0144.html for the rules about naming packages.

**Review by clalancette — CHANGES_REQUESTED** [review]
Besides the name of the package, please open separate PRs for the different distributions.  It is easier to get them in that way.

File: foxy/distribution.yaml (hunk 1 above)
Comment (suurjaak) [reply foxy/distribution.yaml:5713]: Understandable.. I assume the package name in the ROS index does not necessarily need to match the package name in its source repository?

Would `python3-rosros` be acceptable for the ROS index?

Comment (tfoote) [reply foxy/distribution.yaml:5713]: No, that does not resolve any of the issues with the name. Please read the naming guidelines linked above. The name should be at least unique and preferably informative about the contents. Repeating the name of the larger project name twice(which is explicitly said not to use in your package name "A package name should not contain ros as it is redundant.") as a name does not help anyone know what the package is or what it does.

File: rosdep/python.yaml [hunk 2]
```diff
@@ -8800,6 +8800,19 @@ python3-rospkg-modules:
     pip:
       packages: [rospkg]
   ubuntu: [python3-rospkg-modules]
+python3-rosros-pip:
```
Comment (tfoote) [inline rosdep/python.yaml:8803]: Please choose if you're going to be a rosdep or as a native ROS package.

**Review by tfoote — CHANGES_REQUESTED** [review]

**Comment by nuclearsandwich** [comment]
@suurjaak since this PR hasn't been updated recently I am going to close it. Feel free to re-open it when you have time to address the given feedback. If you open a new PR for the same project with an updated name please link back to this pull request so the discussion history is preserved.

**Pull request closed without merging**